clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/view/MaterializedViewWriter.java

```diff
@@ -63,16 +65,32 @@ public CompletableFuture<PubSubProduceResult> processRecord(
       int newValueSchemaId,
       int oldValueSchemaId,
       GenericRecord replicationMetadataRecord) {
-    return processRecord(newValue, key, newValueSchemaId);
+    return processRecord(newValue, key, newValueSchemaId, false);
   }
 
+  /**
+   * Before we have proper chunking support for view writers we assume that even when chunking is enabled the actual
+   * k/v will not be chunked. This way we don't need to worry about how to ensure all the chunks are forwarded to the
+   * same view partition during NR's pass-through mode. Proper chunking support can leverage message footer populated
+   * by the CompositeVeniceWriter in VPJ (write to views first and then VT) to figure out which view partition to
+   * forward the chunks to. Another alternative is trigger the view writer write upon receiving the manifest, and we
+   * will assemble and re-chunk.
+   */
   @Override
-  public CompletableFuture<PubSubProduceResult> processRecord(ByteBuffer newValue, byte[] key, int newValueSchemaId) {
+  public CompletableFuture<PubSubProduceResult> processRecord(
+      ByteBuffer newValue,
+      byte[] key,
+      int newValueSchemaId,
+      boolean isChunkedKey) {
+    byte[] viewTopicKey = key;
+    if (isChunkedKey) {
```

Comment:
Alternatively we could add a new API in the `VeniceWriter` to perform pass through for k/v but regenerates DIV. This would be useful later if we do decide to take the approach of forwarding chunks based on pre-populated message footer for actual large value chunking support.